### PR TITLE
correctly handling NaN values

### PR DESCRIPTION
**Pull request by alexreardon**

### CODE REVIEW

**Comment by alexreardon** [comment]
#101

**Comment by alexreardon** [comment]
What do you think @Ayub-Begimkulov

**Comment by alexreardon** [comment]
My own implementation of #107

File: README.md [hunk 1]
```diff
@@ -59,6 +59,57 @@ yarn add memoize-one
 npm install memoize-one --save
 ```
 
+## Function argument equality
```
Comment (alexreardon) [inline README.md:62]: I also added a lot more detail about the function argument equality

File: src/are-inputs-equal.ts [hunk 2]
```diff
@@ -1,3 +1,25 @@
+// Number.isNaN as it is not supported in IE11 so conditionally using ponyfill
+// Using Number.isNaN where possible as it is ~10% faster
+
+const safeIsNaN =
+  Number.isNaN ||
+  function ponyfill(value: unknown): boolean {
+    return typeof value === 'number' && value !== value;
+  };
```
Comment (Ayub-Begimkulov) [inline src/are-inputs-equal.ts:8]: Nice catch, haven't even thought about the performance difference.

**Comment by Ayub-Begimkulov** [comment]
> What do you think @Ayub-Begimkulov?

Looks great! 

Closing my PR in favor of this.

**Pull request merged**